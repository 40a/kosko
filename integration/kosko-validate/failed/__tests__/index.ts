import execa from "execa";
import { dirname } from "path";
import { runCLI } from "../../../run";

const testDir = dirname(__dirname);

let result: execa.ExecaReturns;

beforeEach(async () => {
  result = await runCLI(["validate"], {
    cwd: testDir,
    reject: false
  });
});

test("should return status code 1", () => {
  expect(result.code).toEqual(1);
});

test("should print the error", () => {
  expect(result.stderr).toStartWith(
    "ValidationError: data.spec should have required property 'containers'"
  );
});
