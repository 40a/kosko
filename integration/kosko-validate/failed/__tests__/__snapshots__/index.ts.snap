// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should print the error 1`] = `
"ValidationError: data.spec should have required property 'containers'
Path: <PROJECT_ROOT>/integration/kosko-validate/failed/components/foo.js
Index: 0
    at Object.<anonymous> (<PROJECT_ROOT>/packages/generate/dist/generate.js:53:35)
    at Generator.next (<anonymous>)
    at fulfilled (<PROJECT_ROOT>/packages/generate/node_modules/tslib/tslib.js:104:62)
"
`;
